ietf-layer0-types-ext.yang

```diff
@@ -51,4 +51,397 @@ module ietf-layer0-types-ext {
+    description
+      "base line-coding class";
+    reference 
+      "ITU-T G.698.2-201811 section 7"   
+  }  
+
+  identity line-coding-NRZ-2p5G {
+    base line-coding;
+    description 
+      "ITU-T G.698.2-201811 section 7 table 8-1 "   
+  }
+
+  identity line-coding-NRZ-OTU1 {
+    base line-coding;
+    description
+      "ITU-T G.698.2-201811 section 7 table 8-2 "   
+  }
+
+  identity line-coding-NRZ-10G {
+    base line-coding;
+    description
+      "ITU-T G.698.2-201811 section 7 table 8-3/8-5 "   
+  }
+
+  identity line-coding-NRZ-OTU2 {
+    base line-coding;
+    description
+      "ITU-T G.698.2-201811 section 7 table 8-4/8-6 "   
+
+  }
+
+// typedefs                
+
+  typedef operational-mode {
+    type string;
+    description
+      "Vendor-specific mode that guarantees 
+      interoperability.";
+    
+  }
+
+// temporary defined here for disalignment with 
+//ietf-layer0-types module
+  typedef standard-mode {
+    type string;
+    description
+      "ITU-T G.698.2 standard mode that guarantees 
+       interoperability.
+       It must be an string with the following format:
+       B-DScW-ytz(v) where all these attributes 
+       are conformant
+       to the ITU-T recomendation";
+    reference "ITU-T G.698.2 (11/2018)";
+  }
+
+  typedef frequency-thz {
+      type decimal64 {
+        fraction-digits 6;
+      }
+      units THz;
+      description
+        "The DWDM frequency in THz, e.g., 193.112500";
+      reference
+        "RFC6205: Generalized Labels for
+        Lambda-Switch-Capable (LSC) Label Switching Routers";
+  }
+  typedef frequency-ghz {
+      type decimal64 {
+        fraction-digits 3;
+      }
+      units GHz;
+      description
+        "The DWDM frequency in GHz, e.g., 193112.500";
+      reference
+        "RFC6205: Generalized Labels for
+        Lambda-Switch-Capable (LSC) Label Switching Routers";
+  }    
+
+// GROUPINGS  
+
+
+ /* supported inverse multiplexing capabilities such as max. OTSiG:OTSi cardinality 
+ It is a transponder attribute not tranceiver*/
+
+/*    leaf multiplexing-cap {
+      type uint32;
+    config false;
+      description "supported inverse multiplexing capabilities
+       such as max. OTSiG:OTSi cardinality";
+    }  */
+
+  
+
+
+/* This grouping represent the list of attributes related to */
+/* optical impairment limits for each operational mode */
+/*(min OSNR, max PMD, max CD, max PDL, Q-factor limit, etc.)*/
+/* It can be utilized also in case of "explicit-mode" */
+
+  grouping common-tranceiver-capabilities-organizational-mode {
+    description "Attributes capabilities related to operational 
+    modes of an optical tranceiver";
+
+
+    leaf line-coding-bitrate {
+      type identyref {
+        base line-coding;
+      }
+      config false;
+      description "Bit rate/line coding of optical tributary signal"
+      reference 
+        "ITU-T G.698.2 section 7.1.2"
+    }
+    leaf max-polarization-mode-dispersion {

```

Comment:
I thought we had an agreement on an triplet CD, PMD Penalty here
https://github.com/ietf-ccamp-wg/draft-ietf-ccamp-optical-impairment-topology-yang/pull/35
Is this a complementary view ?

Reply:
Correct. Need to update , adding the list of triplets:

list chromatic-and-polarization-dispersion-penalty {
      description
          "Optional penalty associated with a given accumulated CD and PMD.
          This list of triplet cd, pmd, penalty can be used to sample 
          the function penalty = f(CD, PMD).";
      config false;
      uses cd-pmd-penalty ;
     }